src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
```

Comment:
JIT fail. In the class case it partially eliminates some the code inside the loop, in the struct case it insists on copying the struct fields to `obj1` and `obj2`. If you change both loop bodies to 
```C#
if (item.Object1 == item.Object2)
    Console.WriteLine("eq");
```
you'll get similar results for both struct and class. The class case is still a tiny bit faster due to other JIT issues but the difference is small enough that avoiding some heap allocations seems preferable.

That's with RyuJIT, I haven't tested with the x86 JIT. Is this code ever supposed to run on x86? As is now CoreCLR doesn't quite support x86.